pkg/controller/migmigration/stage.go

```diff
@@ -147,7 +147,7 @@ func (t *Task) createStagePods(client k8sclient.Client, stagePods StagePodList)
 
 func (t *Task) listStagePods(client k8sclient.Client) (StagePodList, error) {
 	podList := corev1.PodList{}
-	options := k8sclient.MatchingLabels(t.stagePodLabels())
+	options := k8sclient.MatchingLabels(t.stagePodIdentifierLabel())
```

Comment:
Wouldn't the question of "list all stage pods" vs. "list stage pods for this MigPlan" both be needed, depending on context? If we're trying to find stage pods created for this migration for the purpose of using them, shouldn't we use the correlation label? We wouldn't want to rely on an old stage pod created for a different MigPlan (which should have been deleted by now anyway). The "list all stage pods regardless of migplan" would be mainly used for deletion. Also, isn't this context in this function not a single-namespace-scoped list? i.e. it will list all stage pods in all namespaces. We definitely want the migplan filtering there I'd think, or we might end up seeing pods for other migplans.

Reply:
Good points. I will make some changes to:

1. Continue using the correlation label based stage pod list where we can
2. For cleanup purposes, have a separate function that constructs a list of stage pods _in the namespaces selected for migration_

Reply:
@sseago I addressed your comments in the latest change.